README.md

```diff
@@ -1,367 +1,654 @@
-        ]),
-        'payload' => null,
-    ], [
-        'subscription' => Subscription::create([
-            'endpoint' => 'https://example.com/other/endpoint/of/another/vendor/abcdef...',
-            'publicKey' => '(stringOf88Chars)',
-            'authToken' => '(stringOf24Chars)',
-            'contentEncoding' => 'aesgcm', // one of PushManager.supportedContentEncodings
-        ]),
-        'payload' => '{msg:"test"}',
-    ], [
-          'subscription' => Subscription::create([ // this is the structure for the working draft from october 2018 (https://www.w3.org/TR/2018/WD-push-api-20181026/) 
-              "endpoint" => "https://example.com/other/endpoint/of/another/vendor/abcdef...",
-              "keys" => [
-                  'p256dh' => '(stringOf88Chars)',
-                  'auth' => '(stringOf24Chars)'
-              ],
-          ]),
-          'payload' => '{"msg":"Hello World!"}',
-      ],
-];
-
-$webPush = new WebPush();
-
-// send multiple notifications with payload
-foreach ($notifications as $notification) {
-    $webPush->queueNotification(
-        $notification['subscription'],
-        $notification['payload'] // optional (defaults null)
-    );
-}
+//PSR-18 HTTP client
+/** @var Psr\Http\Client\ClientInterface $client */
+$client = '…';
 
-/**
- * Check sent results
- * @var MessageSentReport $report
- */
-foreach ($webPush->flush() as $report) {
-    $endpoint = $report->getRequest()->getUri()->__toString();
-
-    if ($report->isSuccess()) {
-        echo "[v] Message sent successfully for subscription {$endpoint}.";
-    } else {
-        echo "[x] Message failed to sent for subscription {$endpoint}: {$report->getReason()}";
-    }
-}
+//PSR-17 request factory
+/** @var Psr\Http\Message\RequestFactoryInterface $requestFactory */
+$requestFactory = '…';
 
-/**
- * send one notification and flush directly
- * @var MessageSentReport $report
- */
-$report = $webPush->sendOneNotification(
-    $notifications[0]['subscription'],
-    $notifications[0]['payload'] // optional (defaults null)
+// Extension Manager is detailed later
+$extensionManager = ExtensionManager::create();
+
+// The WebPush service
+$webPush = new WebPush(
+    $client,          //PSR-18 HTTP client
+    $requestFactory,  //PSR-17 request factory
+    $extensionManager //Will be explained later
 );
+/* You can also use the following method
+$webPush = WebPush::create($client, $requestFactory, $extensionManager);
+*/
+
+
+// In this example, we consider you already received a subscription object
+// from the user agent
+$subscription = Subscription::createFromString('{"endpoint":"…"}');
+
+// We create a notification
+// No payload or other information at this stage.
+// These features will be explained later
+$notification = Notification::create();
+
+// We send the notification
+$statusReport = $webPush->send($notification, $subscription);
+
+//The status report is either a Success or a Failure one.
```

Comment:
How do you access the fields of StatusReportFailure from $statusReport if it is not a success?

Reply:
You have a full access to the request and response objects with `getRequest()` and `getResponse()` respectively.